compiler/rustc_mir_transform/src/inline.rs

```diff
@@ -453,6 +458,15 @@ impl<'tcx> Inliner<'tcx> {
                 if ty.needs_drop(tcx, self.param_env) && let Some(unwind) = unwind {
                         work_list.push(unwind);
                     }
+            } else if callee_attrs.instruction_set != self.codegen_fn_attrs.instruction_set
+                && matches!(term.kind, TerminatorKind::InlineAsm { .. })
```

Comment:
Should this go into `CostChecker::visit_terminator`?

Reply:
Perhaps. I just saw a thing that seemed to be iterating the function body and put the check where it seemed to fit.